inc/functions.php

```diff
@@ -1807,6 +1807,9 @@ function is_moderator($fid=0, $action="", $uid=0)
 	}
 
 	$user_perms = user_permissions($uid);
+
+	$plugins->run_hooks("is_moderator");
```

Comment:
It would probably be good to pass at least the function arguments and the user permissions.

As suggested by @lairdshaw I've also added a way to allow plugins to return a result that will short-circuit the rest of the function.

```suggestion
	$hook_args = array(
		'fid' => $fid,
		'action' => $action,
		'uid' => $uid,
		'user_perms' => $user_perms',
	);

	$plugins->run_hooks("is_moderator", $hook_args);
	
	if (isset($hook_args['is_moderator']) && $hook_args['is_moderator'] === true) {
		return true;
	}
```